### PR TITLE
Roll out Python wheel build and publishing

### PR DESCRIPTION
**Pull request by t0yv0**
Fixes https://github.com/pulumi/pulumi-aws-native/issues/917

As part of https://github.com/pulumi/home/issues/2883 we are enabling:

- generating pyproject.toml instead of setup.py
- building wheels alongside source distributions
- making sure both wheels and source distributions are published on pypi

If all goes well this feature-* PR should result in a PyPI version that can be checked to have both sdist and wheel distros published.

### CODE REVIEW

**Comment by t0yv0** [comment]
Will try to land https://github.com/pulumi/pulumi-aws-native/pull/1079 first so those changes are not part of this PR.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -457,6 +457,9 @@ jobs:
       run: ./ci-scripts/ci/publish-tfgen-package ${{ github.workspace }}
       env:
         NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+        # See https://github.com/pulumi/scripts/pull/138/files
+        # Possible values: "all", "wheel".
+        PYPI_PUBLISH_ARTIFACTS: all
```
Comment (t0yv0) [inline .github/workflows/build.yml:462]: Need to add this to ci-mgmt, noting.

**Comment by t0yv0** [comment]
And there we have it https://pypi.org/project/pulumi-aws-native/0.75.1a1694201951/#files

Observe both wheel and source distributions being published to PyPI.

**Comment by t0yv0** [comment]
@pulumi/providers  🙏

File: Makefile [hunk 2]
```diff
@@ -80,17 +80,24 @@ build_nodejs::
 		sed -i.bak -e "s/\$${VERSION}/$(VERSION)/g" ./bin/package.json
 
 generate_python::
+	# Delete files not tracked in Git
+	cd sdk/python/ && git clean -fxd
```
Comment (EronWright) [inline Makefile:84]: Consider bailing if the folder is dirty.

**Review by mjeffryes — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -80,17 +80,24 @@ build_nodejs::
...
         cp ../../README.md . && \
-        python3 setup.py clean --all 2>/dev/null && \
         rm -rf ./bin/ ../python.bin/ && cp -R . ../python.bin && mv ../python.bin ./bin && \
-        sed -i.bak -e 's/^VERSION = .*/VERSION = "$(PYPI_VERSION)"/g' -e 's/^PLUGIN_VERSION = .*/PLUGIN_VERSION = "$(VERSION)"/g' ./bin/setup.py && \
-        rm ./bin/setup.py.bak && \
-        cd ./bin && python3 setup.py build sdist
+        sed -i.bak -e 's/^  version = .*/  version = "$(PYPI_VERSION)"/g' ./bin/pyproject.toml && \
+        rm ./bin/pyproject.toml.bak && \
+	python3 -m venv venv && \
+	./venv/bin/python -m pip install build && \
```
Comment (EronWright) [inline Makefile:98]: Seems you are taking the assumption that executing python from the venv dir is akin to activation. I wonder if there's a more formal way to do this.

**Review by EronWright — APPROVED** [review]

**Comment by t0yv0** [comment]
Backlogged some homework https://github.com/pulumi/home/issues/3096

**Pull request merged**